build-tools/src/main/resources/checkstyle.xml

```diff
@@ -173,8 +175,6 @@
         <!-- See http://checkstyle.sf.net/config_modifiers.html -->
         <module name="ModifierOrder"/>
         <module name="RedundantModifier"/>
-        <module name="FileContentsHolder"/>
```

Comment:
Why is this removed? From the docs it seems this is needed if used in conjunction with `SuppressWithNearbyCommentFilter`: https://checkstyle.sourceforge.io/version/6.18/config_filters.html#SuppressWithNearbyCommentFilter

>  Usage: This filter only works in conjunction with a FileContentsHolder, since that check makes the suppression comments in the Java files available. A configuration that includes this filter must configure FileContentsHolder as a child module of TreeWalker.

Reply:
According to this release note: https://checkstyle.sourceforge.io/releasenotes.html#Release_8.2, this is not necessary again after version 8.2. Here we use 8.24 to make it consist with the latest setup on the azure-sdk-for-java-repo.